analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyNullableValueAccess.vb

```diff
@@ -0,0 +1,503 @@
+                    Console.WriteLine(foo.Value.ToString()) ' Noncompliant, FP: unreachable
+                End If
+            End If
+        End If
+    End Sub
+
+    Public Sub NestedIsolated2(ByVal value As DateTime?)
+        If foo = value Then                                 ' Relationship is added here
+            If Not value.HasValue Then
+                If foo Is Nothing Then
+                    Console.WriteLine(foo.Value.ToString()) ' Noncompliant
+                End If
+            End If
+        End If
+    End Sub
+
+    Public Sub NestedCombined(ByVal value As DateTime?)
+        If foo = value Then                                 ' Relationship is added here
+            If value.HasValue Then
+                If foo.HasValue Then
+                    Console.WriteLine(foo.Value.ToString()) ' Compliant
+                End If
+                If Not foo.HasValue Then
+                    Console.WriteLine(foo.Value.ToString()) ' Noncompliant, FP: unreachable
+                End If
+                If foo Is Nothing Then
+                    Console.WriteLine(foo.Value.ToString()) ' Compliant, unreachable
+                End If
+                If foo IsNot Nothing Then
+                    Console.WriteLine(foo.Value.ToString()) ' Compliant
+                End If
+            Else
+                If foo.HasValue Then
+                    Console.WriteLine(foo.Value.ToString()) ' Compliant, unreachable
+                End If
+                If Not foo.HasValue Then
+                    Console.WriteLine(foo.Value.ToString()) ' Noncompliant
+                End If
+                If foo Is Nothing Then
+                    Console.WriteLine(foo.Value.ToString()) ' Compliant, unreachable
+                End If
+                If foo IsNot Nothing Then
+                    Console.WriteLine(foo.Value.ToString()) ' Compliant, unreachable
+                End If
+            End If
+        End If
+    End Sub
+End Class
+
+Class Casts
+    Private Sub Downcast1(ByVal i As Integer?)
+        Dim x = CInt(i) ' Compliant
+        i = Nothing
+        x = CInt(i)     ' FIXME Non-compliant
+        x = CInt(i)     ' Compliant, unreachable
+    End Sub
+
+    Private Sub Downcast2(ByVal i As Integer?)
+        i = 42
+        Dim x = CInt(i) ' Compliant
+    End Sub
+
+    Private Sub Downcast3()
+        Dim i As Integer? = Nothing
+        Dim x = i.Value ' Noncompliant
+        x = CInt(i)     ' Compliant, unreachable
+    End Sub
+
+    Private Sub Upcast1(ByVal i1 As Integer?, ByVal i2 As Integer?)
+        Dim x = (CType(Nothing, Integer?)).Value    ' FIXME Non-compliant
+    End Sub
+
+    Private Sub Upcast2(ByVal i1 As Integer?, ByVal i2 As Integer?)
+        Dim x = (CType(Nothing, Integer?)).Value    ' FIXME Non-compliant
+        i2 = Nothing
+        x = (CType(i2, Integer?)).Value             ' Compliant, unreachable
+    End Sub
+
+    Private Sub AsOperator1(ByVal i As Integer?)
```

Comment:
Since VB.NET lacks of a proper `as` operator, spelled that way, I would rename this method and the following one to match VB.NET specifics.